### DISCUSSION TITLE
spring-boot-2/4-6-multi-spring-data-jpa

### DISCUSSION THREAD

**Discussion by giscus** [dyc87112/giscus-didispace#28] [Announcements]
# spring-boot-2/4-6-multi-spring-data-jpa

Spring Boot 2.x基础教程：Spring Data JPA的多数据源配置 上一篇我们介绍了在使用JdbcTemplate来做数据访问时候的多数据源配置实现。接下来我们继续学习如何在使用Spring Data JPA的时候，完成多数据源的配置和使用。 添加多数据源的配置 先在Spring Boot的配置文件application.properties中设置两个你要链接的数据库配置，比如这样：

https://www.didispace.com/spring-boot-2/4-6-multi-spring-data-jpa.html

**Comment by liukaiK** [comment]
这两个数据源连接的都是MySQL，
如果一个连接MySQL，另一个连接postgresql的话。下面的配置就不够用了吧。
```properties
spring.datasource.primary.jdbc-url=jdbc:mysql://localhost:3306/test1
spring.datasource.primary.username=root
spring.datasource.primary.password=123456
spring.datasource.primary.driver-class-name=com.mysql.cj.jdbc.Driver

spring.datasource.secondary.jdbc-url=jdbc:mysql://localhost:3306/test2
spring.datasource.secondary.username=root
spring.datasource.secondary.password=123456
spring.datasource.secondary.driver-class-name=com.mysql.cj.jdbc.Driver

```
还得配置数据库方言呢

**Reply by liukaiK** [reply]
还有这样配置的话，IDEA全是黄线。因为这些不是SpringBoot约定的配置，这个黄线问题有什么好的解决方式么？
